modules/core/shared/src/main/scala/lucuma/core/model/Target.scala

```diff
@@ -3,16 +3,32 @@
 
 package lucuma.core.model
 
-import cats._
-
-import lucuma.core.math.ProperMotion
+import scala.collection.immutable.SortedMap
 
-import monocle.macros.Lenses
+import cats._
+import cats.implicits._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.string.NonEmptyString
+import lucuma.core.enum.MagnitudeBand
+import lucuma.core.math._
+import monocle.Lens
+import monocle.Optional
+import monocle.Traversal
+import monocle.function.Each.mapEach
+import monocle.function.FilterIndex
+import monocle.macros.GenLens
+import monocle.std.either._
+import monocle.std.option
 
 /** A target of observation. */
-@Lenses final case class Target(name: String, track: Either[EphemerisKey, ProperMotion])
+final case class Target(
+  name:       NonEmptyString,
+  catalogId:  Option[NonEmptyString],
+  track:      Either[EphemerisKey, ProperMotion],
+  magnitudes: SortedMap[MagnitudeBand, Magnitude]
```

Comment:
This assumes we only have a single value per band.

Reply:
Seems fine. I think this is out the window anyway when we get time-varying magnitudes.